tests/integration/test_deploy.py

```diff
@@ -0,0 +1,106 @@
+from typing import List
+import pytest
+from pytest_operator.plugin import OpsTest
+from juju.unit import Unit
+import logging
+
+log = logging.getLogger(__name__)
+
+certs = [
+    "ETCDCTL_KEY_FILE=/var/snap/etcd/common/client.key",
+    "ETCDCTL_CERT_FILE=/var/snap/etcd/common/client.crt",
+    "ETCDCTL_CA_FILE=/var/snap/etcd/common/ca.crt",
+    "ETCDCTL_KEY=/var/snap/etcd/common/client.key",
+    "ETCDCTL_CERT=/var/snap/etcd/common/client.crt",
+    "ETCDCTL_CACERT=/var/snap/etcd/common/ca.crt",
+]
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    charm = await ops_test.build_charm(".")
+    await ops_test.model.deploy("cs:~containers/easyrsa-441")
+    resources = {"snapshot": "snapshot.tar.gz"}
+    await ops_test.model.deploy(charm, resources=resources)
+    await ops_test.model.add_relation("easyrsa:client", "etcd:certificates")
+    await ops_test.model.wait_for_idle(wait_for_active=True, timeout=60 * 60)
+
+
+async def test_status(ops_test):
+    assert ops_test.model.applications["etcd"].units[0].workload_status == "active"
+
+
+async def _get_leader(units: List[Unit]) -> Unit:
+    for unit in units:
+        if await unit.is_leader_from_status():
+            return unit
+
+
+async def test_leader(ops_test: OpsTest):
+    leader = await _get_leader(ops_test.model.applications["etcd"].units)
+    assert leader is not None
```

Comment:
this is really a test of juju, not of etcd.  One of them is going to be the leader whether it likes it or not -- and you've only deployed one unit -- so it's gonna be the leader.  I think this is fruitless and probably not a great test